### ISSUE TITLE
Functional matchers should use more specific interfaces for arguments

### ISSUE THREAD

**Issue by roscopeco** [roscopeco/moxy#10]
**Describe the solution you'd like**
Matchers `matches` et. al. should use more specific functional interfaces (e.g. `Predicate<T>` instead of `Function<T, Boolean>`).

**Additional context**
Low priority, but should be done before 1.0 due to API change.

**Comment by roscopeco** [comment]
Fixed in develop; for 0.92 release.
